### PR TITLE
Add char type

### LINKED ISSUE #4: Add character type
**Syntax**

```
let a = 'a'
let b = 'b'
let ab = 'ab' # wrong
```

**Semantics**

The behavior of `==`, `>`, `>=`, `<`, `<=` operators should be the same as in Haskell.

f.e.

```
'a' == 'a'  # True
'b' > 'a'    # True
'B' > 'a'    # False
```

### PR DESCRIPTION
**Pull request by aitorres**
This PR adds support for the character type:

- Adds both Parser and Interpreter tests for handling char type and char comparisons.
- Adds support for Char in the Language, AST and parsing
- Adds support for Char comparison in interpreter

This PR closes #4.

### CODE REVIEW

File: src/Interpreter/Eval.hs [hunk 1]
```diff
@@ -114,7 +116,8 @@ evalNumCompOp op expr1 expr2 scope = do
   (res2, _) <- evaluate expr2 scope
   case (res1, res2) of
     (Integer val1, Integer val2) -> Right $ (Bool $ op val1 val2, scope)
-    _ -> Left "Can compare only numbers"
+    (Char c1, Char c2) -> Right $ (Bool $ op (toInteger . ord $ c1) (toInteger . ord $ c2), scope)
+    _ -> Left "Can only compare two comparable types"
```
Comment (yurtsiv) [inline src/Interpreter/Eval.hs:121]: Rename this function to `evalCompOp` as `Num` is unnecessary

File: test/Interpreter/EvalSpec.hs [hunk 2]
```diff
@@ -63,6 +64,18 @@ spec = do
...
+    let chars = [r|
+      let a = 'a'
+      let b = 'b'
+
+      a == b
+      a < b
+      a > b
+      a >= b
+      a <= b
+    |]
```
Comment (yurtsiv) [inline test/Interpreter/EvalSpec.hs:83]: You forgot to do `testProgramm`. In order to test everything at once, you can return a list

```     
[
  a == b,
  a < b,
  a > b,
  a >= b,
  a <= b
]
```

File: src/Interpreter/Eval.hs (hunk 1 above)
Comment (aitorres) [reply src/Interpreter/Eval.hs:121]: Done! Thanks for spotting this, I thought I had renamed it hehe

File: test/Interpreter/EvalSpec.hs (hunk 2 above)
Comment (aitorres) [reply test/Interpreter/EvalSpec.hs:83]: Changed, thanks for the idea of testing everything at once through a list. I also added a test for inequality with chars (`a /= b')

**Comment by yurtsiv** [comment]
@aitorres Great, I'm merging!

**Pull request merged**